File: lib/gun.rb
class Gun
  attr_reader :name,
              :quality,
              :damage,
              :notes

  def initialize(data)
    @name    = data[:name]
    @quality = data[:quality] == "N/A" ? nil : data[:quality].delete("Quality Item.png")
    @damage  = data[:damage].nil? ? nil : data[:damage].to_s
    @notes   = data[:notes]
  end
end

File: lib/gun.js
'use strict';

function Gun(data) {
  this.name    = data.Name ? data.Name.toLowerCase() : '';
  this.quality = data.Quality ? data.Quality.replace('Quality Item.png', '').trim() : 'not found';
  this.damage  = data.Damage ? data.Damage.toString().toLowerCase() : 'not found';
  this.notes   = data.Notes ? data.Notes.toLowerCase() : 'not found';
}

module.exports = Gun;
